druid/src/widget/scroll.rs

```diff
@@ -410,9 +515,21 @@ impl<T: Data, W: Widget<T>> Widget<T> for Scroll<T, W> {
             }
         }
 
+        if let Event::Command(ref cmd) = event {
+            if let Some(scroll_to) = cmd.get(SCROLL_TO) {
```

Comment:
Just a remark that this is possible without changing `Scroll`, by using `WidgetExt::controller` to wrap the scroll in a controller that intercepts the command. (That kind of composition is my "ideal" for how druid should work, but as you know I don't have a concrete solution for the command-submission/handler/callback half of that ideal; therefore, this is just a remark, not a request for a change.)

Reply:
Hm, thats different than I would expect. Surely there would be some expectations of commands that widgets support out of the box? Given that it is the main way of interacting with them. It does feel like you might end up having to replicate all of your 'interactive' methods as commands...

Reply:
I did a quick grep. Currently, `TextBox` takes several commands, but not so much for the other built-in widgets. Probably this is partly because we don't have lots of complex built-in widgets and partly because some things that might be slots in (e.g.) qt instead get their behavior from Data (e.g. the text for a `Label`).

Reply:
I do think it is possible that later down the line we will come up with a sane way somehow put all of this "possibly reactive" stuff  in data, and when its not needed seamlessly do the wrapping and unwrapping hygienically. But I haven't quite worked it out yet..

Reply:
I've thought about it and I don't think this holds water. I fully understand the benefits of composition, but being able to tell a scroll area to scroll is not at all specialised. Having every app come up with their own series of adhoc commands for widgets is not a good idea. Commands literally are the way you communicate with widgets after they have been boxed up in another widget and really every post construction api method should have a command counterpart.

Reply:
> really every post construction api method should have a command counterpart

I realize you're probably sick of seeing me write this, but I remain very much unconvinced about this part. In my admittedly limited experience with druid, post-construction APIs are only used locally (like in the example here, where you use it to communicate between siblings). If we had a better story for this kind of local communication, I think the non-local case would be rare enough that ad-hoc controllers would be good enough.

Either way, I'd be interested in revisiting this conversation when we all have more experience with composite widgets. As far as this PR goes, I'm not going to complain about it...